src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Rfc3161TimestampTokenFactory.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+
+namespace NuGet.Packaging.Signing
+{
+    internal sealed class Rfc3161TimestampTokenFactory
+    {
+        public static IRfc3161TimestampToken Create(
+            IRfc3161TimestampTokenInfo tstInfo,
+            X509Certificate2 signerCertificate,
+            X509Certificate2Collection additionalCerts,
+            byte[] encoded)
+        {
```

Comment:
On each of your Factory classes (i.e.:  `CmsFactory`, `Rfc3161TimestampRequestFactory`, `Rfc3161TimestampTokenFactory.cs`, `Rfc3161TimestampTokenInfoFactory`, etc.) add basic `null` checks.

Reply:
Thanks! Added `null` checks for the four factory classes.